### ISSUE TITLE
Add a miri test to azure ci

### ISSUE THREAD

**Issue by lu-zero** [rust-lang/stdarch#805]
We have to keep in mind that miri can be out of sync compared to the current nightly.

[this snippet](https://github.com/rust-lang/miri/#running-miri-on-ci) should provide the basis for it.

**Comment by sayantn** [comment]
the problem is that most of the repo uses LLVM intrinsics, so a very small subset will actually benefit, and we will have to have `cfg(miri, ignore)` everywhere

**Comment by RalfJung** [comment]
FWIW miri-test-libstd runs some of the stdarch tests these days, see https://github.com/rust-lang/miri-test-libstd/blob/master/run-stdarch-test.sh.

**Comment by sayantn** [comment]
Oh, interesting. Does miri support some LLVM linking now?

**Comment by RalfJung** [comment]
No. We just have implementations for more than 100 LLVM intrinsics inside Miri so we support Intel vector extensions from SSE2 to AVX2. ;)

**Comment by sayantn** [comment]
That seems like quite a few intrinsics to make a miri test worth it. How is the ARM/AArch64 intrinsics support in Miri?

**Comment by RalfJung** [comment]
> How is the ARM/AArch64 intrinsics support in Miri?

It is non-existent.

For x86, stdarch uses the portable `simd_*` intrinsics in many places, making the Miri implementation feasible. For Arm, it relies much more on LLVM intrinsics, so a Miri implementation would be a lot more work -- it's not clear that it would be practical. See https://github.com/rust-lang/stdarch/issues/1659.
